src/rpc/rpc_handler.cpp

```diff
@@ -63,7 +63,9 @@ namespace rpc
             d.cached_to -= 10;
             d.cached_top_hash = hash10;
             d.cached_m10_hash = crypto::null_hash;
-            d.cached_distribution.resize(d.cached_distribution.size() - 10);
+            CHECK_AND_ASSERT_MES(d.cached_distribution.size() >= 10, boost::none, "Cached distribution size does not match cached bounds");
+            for (int p = 0; p < 10; ++p)
```

Comment:
Can this still be a resize with the check above? Or does Gcc complain about that too?

Reply:
I *think* it still complained, but I'd have to reinstall that compiler to double check.